json_annotation/lib/src/json_key.dart

```diff
@@ -149,19 +149,26 @@ class JsonKey {
   /// valid on a nullable enum field.
   final Enum? unknownEnumValue;
 
+  /// If true, enum will be parsed with case-insensitive.
+  /// Specifically, both values will be lower-cased and compared.
+  ///
+  /// Valid only on enum fields with a compatible enum value.
+  final bool caseInsensitive;
+
   /// Creates a new [JsonKey] instance.
   ///
   /// Only required when the default behavior is not desired.
   const JsonKey({
-    @Deprecated('Has no effect') bool? nullable,
+    @Deprecated('Has no effect')
+        bool? nullable,
     this.defaultValue,
     this.disallowNullValue,
     this.fromJson,
     @Deprecated(
       'Use `includeFromJson` and `includeToJson` with a value of `false` '
       'instead.',
     )
-    this.ignore,
+        this.ignore,
```

Comment:
fyi this was done by dart format 🤷

Reply:
You need to be using a dev release of Dart 3.1 for this.